Src/EventCounterCollector/EventCounterCollector/EventCounterListener.cs

```diff
@@ -198,6 +198,19 @@ private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, ob
                     {
                         actualCount = Convert.ToInt32(payload.Value, CultureInfo.InvariantCulture);
                     }
+                    else if (key.Equals("Metadata", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var metadata = payload.Value.ToString();
+                        if (!string.IsNullOrEmpty(metadata))
+                        {
+                            var keyValuePairStrings = metadata.Split(',');
+                            foreach (var keyValuePairString in keyValuePairStrings)
+                            {
+                                var keyValuePair = keyValuePairString.Split(':');
+                                metricTelemetry.Properties.Add(keyValuePair[0], keyValuePair[1]);
```

Comment:
can you check and not add property, if property already have a key with same name?